ocs_ci/ocs/cluster.py

```diff
@@ -1450,9 +1450,14 @@ def check_ceph_osd_tree_after_node_replacement():
     """
     ct_pod = pod.get_ceph_tools_pod()
     osd_tree = ct_pod.exec_ceph_cmd(ceph_cmd="ceph osd tree")
-    if not check_ceph_osd_tree():
-        logger.warning("Incorrect ceph osd tree formation found")
-        return False
+    # We need to add a function to check the ceph osd tree when flexible scaling is enabled.
+    # So currently, when flexible scaling is enabled don't check the Ceph osd tree.
+    # The issue for tracking is https://github.com/red-hat-storage/ocs-ci/issues/4052
+    if not is_flexible_scaling_enabled():
+        if not check_ceph_osd_tree():
+            logger.warning("Incorrect ceph osd tree formation found")
```

Comment:
you can use "and"
if (not is_flexible_scaling_enabled()) and (not check_ceph_osd_tree()):

Reply:
Yes. But in this specific case, I preferred to separate the lines to highlight that only when flexible scaling is not enabled do we need to check the ceph osd tree. When the issue https://github.com/red-hat-storage/ocs-ci/issues/4052 will be fixed, I will remove the line `if not is_flexible_scaling_enabled()`.